enhancements/proposals/1405-alternator/README.md

```diff
@@ -0,0 +1,182 @@
+This allows me to direct traffic directly through a platform LB on a dedicated IP.
+
+### Risks and Mitigations
+
+Not known.
+
+## Design Details
+
+### API changes for ScyllaClusters.scylla.scylladb.com/v1
+
+The following API changes show how we'd approximately extend `scyllaclusters.scylla.scylladb.com/v1.spec.alternator` specification. Only new fields are shown for existing structs.
+
+```golang
+package v1
+
+type TLSCertificateType string
+
+const (
+	TLSCertificateTypeOperatorManaged   TLSCertificateType = "OperatorManaged"
+	TLSCertificateTypeUserManaged       TLSCertificateType = "UserManaged"
+)
+
+type UserManagedTLSCertificateOptions struct {
+	// secretName references a kubernetes.io/tls type secret containing the TLS cert and key.
+	// +kubebuilder:validation:Required
+	SecretName string `json:"secretName"`
+}
+
+type OperatorManagedTLSCertificateOptions struct {
+	// additionalDNSNames represents external DNS names that the certificates should be signed for.
+	// +optional
+	AdditionalDNSNames []string `json:"additionalDNSNames,omitempty"`
+	
+	// additionalIPAddresses represents external IP addresses that the certificates should be signed for.
+	// +optional
+	AdditionalIPAddresses []string `json:"additionalIPAddresses,omitempty"`
+}
+
+type TLSCertificate struct {
+	// type determines the source of this certificate.
+	// +kubebuilder:validation:Enum="OperatorManaged";"UserManaged"
+	Type TLSCertificateType `json:"type"`
+	
+	// userManagedCertificateOptions specify option for certificates manged by users.
+	// +optional
+	UserManagedOptions *UserManagedTLSCertificateOptions `json:"userManagedOptions,omitempty"`
+	
+	// OperatorManagedCertificateOptions specify option for certificates manged by the operator.
+	// +optional
+	OperatorManagedOptions *OperatorManagedTLSCertificateOptions `json:"operatorManagedOptions,omitempty"`
+}
+
+// AlternatorSpec hold options related to Alternator API.
+// +kubebuilder:validation:XValidation:rule="( has(self.port) && self.port != 0) ? self.insecureEnableHTTP != false : true",message="insecureEnableHTTP can't be false when using (insecure) port."
+type AlternatorSpec struct {
+	// insecureEnableHTTP enables serving Alternator traffic also on insecure HTTP port.
+	InsecureEnableHTTP *bool `json:"insecureEnableHTTP,omitempty"`
+	
+	// servingCertificate references a TLS certificate for serving secure traffic.
+	// +kubebuilder:default:="{type:"OperatorManaged"}"
+	// +optional
+	ServingCertificate *TLSCertificate `json:"servingCertificate,omitempty"`
+}
+```
+
+### Certificates
+
+Given we aim to always enable HTTPS we need to provision default serving certificate. Operator will create signer Secret `<sc-name>-local-alternator-serving-ca`, save the CA bundle into ConfigMap `<sc-name>-local-alternator-serving-ca` and create serving Secret `<sc-name>-local-alternator-serving-certs`.
+Serving certs will be valid for 30 days and rotated from 20 day mark.
```

Comment:
This looks unusual low validity and quick rotation. I'm aware of 90 days or 1 year as more regular defaults.

Reply:
This is in line with the other certificates we create.

I've seen 1y to be used only with manual renewals, I think Let's Encrypt prod CA is on 90d but mostly because some of the clients are not automated and this was a compromise between that. Security wise the lower the better. 1 month still gives enough time to deal with issues while giving a good security policy.